### ISSUE TITLE
Automated meetup notifications?

### ISSUE THREAD

**Issue by rwanyoike** [nairobilug/nairobi-bot#1]
A nice enhancement to have would be automated meetup notifications. 

Our meetups (@nairobilug) - every 1st Saturday of the month - are always announced late. If we could automate notifications\* every ~30 days, I think turnout would be better.

\* mailing list (email), twitter and irc. :+1:

**Comment by urbanslug** [comment]
Hello @rwanyoike :smiley_cat: you would like for the bot to send tweets, email and notify on IRC once a month. On  which day of the month? What about the time that this should take place?

I need a little more info.

**Comment by alanorth** [comment]
We know the meeting is on the first Saturday of the month, which you could get programatically—work backwards from there. I assume IRC is easiest (as it's an IRC bot afterall), so perhaps it could announce on IRC ten days before, seven days before, five days before, then three, two, one? :)

**Comment by urbanslug** [comment]
Nice 10, 7, 3,2,1.

It would be fun if we could make it send email and tweet over and above posting on IRC. But email to who? To a list of email addresses? To the mailing list?

**Comment by alanorth** [comment]
E-mail and tweet are harder... because those involve things that are not
IRC, and it's not like read-only fetching a web page that it's already
doing—it would have to use SMTP for sending mail, and then what username,
password, and server does it use? And who does it send to? Same questions
for tweeting. I don't think the IRC bot should send mail or tweet.

**Comment by rwanyoike** [comment]
Sure, lets stick to irc. What do you think @urbanslug, make use of a timer library?

**Comment by urbanslug** [comment]
@alanorth Yeah, do one thing and do it well. Staying true to the Unix philosophy.
@rwanyoike  Yeah, can be done :smile_cat:

**Comment by urbanslug** [comment]
To my surprise this isn't as easy to implement as I expected so I shall it shall take a while. See, https://en.wikipedia.org/wiki/Determination_of_the_day_of_the_week

**Comment by alanorth** [comment]
It's good to learn these things, HAHA!

On Tue, Dec 15, 2015 at 8:38 AM, Njagi Mwaniki notifications@github.com
wrote:

> To my surprise this isn't as easy to implement as I expected so I shall it
> shall take a while see,
> https://en.wikipedia.org/wiki/Determination_of_the_day_of_the_week
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/nairobilug/nairobi-bot/issues/1#issuecomment-164674485
> .

## 

Alan Orth
alan.orth@gmail.com
https://alaninkenya.org
https://mjanja.ch
"In heaven all the interesting people are missing." ―Friedrich Nietzsche
GPG public key ID: 0x8cb0d0acb5cd81ec209c6cdfbd1a0e09c2f836c0

**Comment by urbanslug** [comment]
I just thought that it could query the system date and use that to notify users. Why didn't this simple solution come to mind? :smiley_cat:

**Comment by rwanyoike** [comment]
@urbanslug are you working on this? I set this repo up https://github.com/nairobilug/hello-notify - we should team up!

**Comment by urbanslug** [comment]
@rwanyoike Interesting that you should ask me about the support of another repo in this one. Uh, I could but the time. What about @oguya ? or @eebrah or @fredmanglis ? Let's see their code for once. :smiley_cat:
